In [ ]:
import pandas as pd
from google.colab import files

In [ ]:
uploaded=files.upload()

Saving Cutlets.csv to Cutlets.csv


In [ ]:
data1=pd.read_csv("Cutlets.csv")

In [ ]:
data1

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

Assume Null hyposthesis as Ho: μ1 = μ2 (There is no difference in diameters of cutlets between two units).

Thus Alternate hypothesis as Ha: μ1 ≠ μ2 (There is significant difference in diameters of cutlets between two units) 2 Sample 2 Tail test applicable

In [ ]:
from scipy import stats
pvalue= stats.ttest_ind(data1['Unit A'],data1['Unit B'])[1]

0.4722394724599501

Pvalue=0.472
alpha=0.05
Here Pvalue > alpha so we need to accept null hypothesis which means of mean of Unit A is equal to Unit B

question 2

In [ ]:
uploaded=files.upload()

Saving LabTAT.csv to LabTAT.csv


In [ ]:
data2=pd.read_csv("LabTAT.csv")

In [ ]:
data2

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [ ]:
#Anova ftest statistics: Analysis of varaince between more than 2 samples or columns
#Assume Null Hypothesis Ho as No Varaince: All samples TAT population means are same
#Thus Alternate Hypothesis Ha as It has Variance: Atleast one sample TAT population mean is different

In [ ]:
p_value=stats.f_oneway(data2['Laboratory 1'],data2['Laboratory 2'],data2['Laboratory 3'],data2['Laboratory 4'])

In [ ]:
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [ ]:
# AS P_value<alpha(0.05)
#Accept NULL Hypothesis

In [ ]:
# QUESTION 3

In [ ]:
uploded=files.upload()

Saving BuyerRatio.csv to BuyerRatio (1).csv


In [ ]:
data3=pd.read_csv("BuyerRatio.csv")

In [ ]:
data3

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
#Chi2 contengency independence test
#Assume Null Hypothesis as Ho: Independence of categorical variables (male-female buyer rations are similar across regions (does not vary and are not related)
#Thus Alternate Hypothesis as Ha: Dependence of categorical variables (male-female buyer rations are NOT similar across regions (does vary and somewhat/significantly related)

In [ ]:
# Make dimensional array
import numpy as np
obs=np.array([[50,142,131,70],[435,1523,1356,750]])
obs

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [ ]:
from scipy.stats import chi2_contingency
chi2_contingency(obs)# o/p is (Chi2 stats value, p_value, df, expected obsvations)

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [ ]:
#p_value=0.66 > alpha=0.05
# accept null hypothesis

In [ ]:
# Question 4

In [ ]:
uploaded=files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm.csv


In [ ]:
data4=pd.read_csv("Costomer+OrderForm.csv")

In [ ]:
data4

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [ ]:
data4['Phillippines'].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [ ]:
data4['Indonesia'].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [ ]:
data4['Malta'].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [ ]:
data4['India'].value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [ ]:
# Make a Contiingency table
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [ ]:
chi2_contingency(obs)

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

In [ ]:
#p_value=0.277 > alpha(0.05)
# Accept Null Hypothesis